WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2440,6 +2442,27 @@ static GSourceFuncs _handlerIntervention =
                                             activeURL.c_str()));
             }
 
+            if (!isWebProcessResponsive && _state == PluginHost::IStateControl::SUSPENDED)
+            {
+                SYSLOG(Logging::Notification, (_T("Killing unresponsive suspended WebProcess, pid=%u, reply num=%d(max=%d), url=%s\n"),
+                                            webprocessPID, _unresponsiveReplyNum, kWebProcessUnresponsiveReplyDefaultLimit,
+                                            activeURL.c_str()));
+                if (_unresponsiveReplyNum <= kWebProcessUnresponsiveReplyDefaultLimit)
+                {
+                    _unresponsiveReplyNum = kWebProcessUnresponsiveReplyDefaultLimit;
+                    Logging::DumpSystemFiles(webprocessPID);
+                    if (syscall(__NR_tgkill, webprocessPID, webprocessPID, SIGFPE) == -1)
```

Comment:
Please use the core functionality for this. Create a ProcessInfo with the webprocessPID and call Dump on the object.

Reply:
@MFransen69 ProcessInfo::Dump doesn't work as desired in this case. It uses kill which sends signal to a process ground( any thread can handle that signal). The goal here is to send SIGFPE to a UI thread, hence the use of tgkill.